go/enclave/components/rollup_consumer.go

```diff
@@ -143,43 +136,20 @@ func (rc *rollupConsumerImpl) extractRollups(br *common.BlockAndReceipts, blockR
 	return rollups
 }
 
-// IsGenesis indicates whether the rollup is the genesis rollup.
-// todo (#718) - Change this to a check against a hardcoded genesis hash.
-func IsGenesis(rollupHeader *common.RollupHeader) bool {
-	return rollupHeader.Number.Cmp(big.NewInt(int64(common.L2GenesisHeight))) == 0
-}
-
 // Validates and stores the rollup in a given block. Returns nil, nil when no rollup was found.
 // todo (#718) - design a mechanism to detect a case where the rollup doesn't contain any batches (despite batches arriving via P2P)
 func (rc *rollupConsumerImpl) processRollup(block *types.Block, rollup *common.ExtRollup) error {
-	latestRollup, err := getLatestRollupBeforeBlock(block, rc.storage, rc.logger)
-	if err != nil && !errors.Is(err, db.ErrNoRollups) {
-		return fmt.Errorf("unexpected error retrieving latest rollup for block %s. Cause: %w", block.Hash(), err)
-	}
-
-	// If this is the first rollup we've ever received, we check that it's the genesis rollup.
-	if latestRollup == nil && !IsGenesis(rollup.Header) {
-		return fmt.Errorf("received rollup with number %d but no genesis rollup is stored", rollup.Header.Number)
-	}
-
-	if err = rc.checkRollupsCorrectlyChained(rollup.Header, latestRollup); err != nil {
-		return fmt.Errorf("rollup was not correctly chained. height=%d hash=%s Cause: %w",
-			rollup.Header.Number, rollup.Hash(), err)
-	}
-
-	// todo (@matt) - store batches from the rollup (important during catch-up)
-
 	// do we need to store the entire rollup?
 	// Should be enough to store the header and the batches
-	if err = rc.storage.StoreRollup(rollup); err != nil {
+	if err := rc.storage.StoreRollup(rollup); err != nil {
 		// todo (@matt) - this seems catastrophic, how do we recover the lost rollup in this case?
 		return fmt.Errorf("could not store rollup. Cause: %w", err)
 	}
 
 	// we record the latest rollup published against this L1 block hash
 	rollupHash := rollup.Header.Hash()
 	blockHash := block.Hash()
-	err = rc.storage.UpdateHeadRollup(&blockHash, &rollupHash)
+	err := rc.storage.UpdateHeadRollup(&blockHash, &rollupHash)
```

Comment:
'head rollup' still exists for now, as a temporary stepping stone? Eventually I guess rather than head rollup it'll be like `rollupsPublishedUpToSeqNumber` or something.

Reply:
I don't think we should need the concept of rollup at all.
Maybe just as a source of data.

Reply:
This is all a stepping stone, next PR I'm going to change how rollups are tracked and created.